scipy/stats/_sobol.pyx

```diff
@@ -29,15 +34,64 @@ cdef extern from *:
 _MAXDIM = MAXDIM_DEFINE
 _MAXDEG = MAXDEG_DEFINE
 
-cdef cnp.uint64_t poly[MAXDIM]
-cdef cnp.uint64_t vinit[MAXDIM][MAXDEG]
+_poly_dict = {}
+_vinit_dict = {}
+
+
+def get_poly_vinit(kind, dtype):
```

Comment:
This looks cleaner than calling _initialize_direction_numbers in the Python class.